.checkpatch.conf

```diff
@@ -21,6 +21,7 @@
 --ignore MINMAX
 --ignore MULTISTATEMENT_MACRO_USE_DO_WHILE
 --ignore NETWORKING_BLOCK_COMMENT_STYLE
+--ignore NEW_TYPEDEFS
```

Comment:
I think we first need to discuss whether this is a rule that we want to keep enforcing.

IMHO, avoiding unnecessary typedefs (e.g. simple struct typedefs), makes sense because, well ..., they are unnecessary. Keeping this rule would help prevent people from defining typedefs unless they are _absolutely necessary_. Also have a look at [the rationale from the Linux kernel coding style](https://www.kernel.org/doc/html/v4.10/process/coding-style.html#typedefs).

If we plan to keep enforcing this even partially, we should consider a different approach (e.g. keeping a list of allowed typedefs to suppress the warning).